### DISCUSSION TITLE
Allow using previous properties in declaring future properties on an object initializer

### DISCUSSION THREAD

**Discussion by ekolis** [dotnet/csharplang#2776] [Language Ideas]
For instance:

```csharp
var foo = new Foo
{
  Bar = "hello",
  Baz = Bar.IndexOf("h"), // compile error in current C# version
}
```

**Comment by ekolis** [comment]
A more real life example:

```csharp
var person = new Person
{
  Email = "test@test.com",
  EmailTypeID = 1,
  EmailTypeName = LookupEmailTypeName(EmailTypeID),
}
```

**Comment by ronnygunawan** [comment]
this will break existing code if containing class has EmailTypeID property or local

**Comment by yaakov-h** [comment]
I'm 99% sure this has already been raised in another issue, but cannot for the life of me find that other issue.

**Comment by bricelam** [comment]
I had a similar idea today for a slightly more complex scenario:

```cs
new Table
{
    Columns =
    {
        new Column { Name = "Id" } idColumn // 😈
    },
    PrimaryKey = new PrimaryKey
    {
        Columns = { idColumn }
    }
};
```

**Comment by jnm2** [comment]
@bricelam
Duplicate except for syntax: https://github.com/dotnet/csharplang/issues/400
Similar but more universal: https://github.com/dotnet/csharplang/issues/595
